utility_analysis_new/dp_engine.py

```diff
@@ -142,6 +147,19 @@ def _select_private_partitions_internal(
         # function is no-op.
         return col
 
+    def _extract_columns(
+        self, col,
+        data_extractors: Union[pipeline_dp.DataExtractors,
+                               utility_analysis_new.PreAggregateExtractors]):
+        """Extract columns using data_extractors."""
+        if self._options.pre_aggregated_data:
+            # todo: check data_extractors is PreAggregateExtractors
+            return self._backend.map(
```

Comment:
Can the extraction logic be moved to PreAggregateExtractors ? It seems to belong there conceptually.